<?xml version="1.0"?>
<launch>
  <arg name="use_sim_time" default="false" />
  <arg name="enable_ekf" default="$(env ENABLE_EKF true)"/>

  <!-- <node pkg="controller_manager" exec="spawner" name="controller_spawner" args="racecar_joint_publisher racecar_velocity_controller">
    <param from="$(find-pkg-share racecar_control)/config/control.yaml" />
  </node> -->

  <group if="$(var enable_ekf)">
    <node pkg="robot_localization" exec="ekf_node" name="ekf_node">
      <param name="use_sim_time" value="$(var use_sim_time)" />
      <param from="$(find-pkg-share racecar_control)/config/robot_localization.yaml" />
    </node>
  </group>
  <!-- Debug log level -->
  <node pkg="ackermann_mux" exec="ackermann_mux" name="ackermann_mux" output="screen">
    <param from="$(find-pkg-share racecar_control)/config/ackermann_mux.yaml" />
    <remap from="throttle_cmd_out" to="throttle/velocity/command" />
    <remap from="steering_cmd_out" to="steering/position/command" />
  </node>

  <!-- <node pkg="racecar_control" exec="cmd_vel_converter" name="cmd_vel_converter">
    <remap from="cmd_vel_in" to="racecar_velocity_controller/cmd_vel_unstamped" />
    <remap from="cmd_vel_out" to="/ackermann_steering_controller/reference" />
  </node> -->
  
</launch>